token/quote.go

```diff
@@ -10,30 +10,53 @@ import (
 
 // QuoteSQLString returns quoted string with SQL string escaping.
 func QuoteSQLString(s string) string {
+	quote := suitableQuote([]byte(s))
+
 	var buf bytes.Buffer
-	buf.WriteByte('"')
-	quoteSQLStringContent(s, &buf)
-	buf.WriteByte('"')
+	buf.WriteRune(quote)
+	quoteSQLStringContent(quote, s, &buf)
+	buf.WriteRune(quote)
 	return buf.String()
 }
 
+func suitableQuote(b []byte) rune {
+	var hasSingle, hasDouble bool
+	for _, b := range b {
+		switch b {
+		case '\'':
+			hasSingle = true
+		case '"':
+			hasDouble = true
+		}
+	}
+	if !hasSingle && hasDouble {
+		return '\''
+	}
+	return '"'
+}
+
 // QuoteSQLString returns quoted string with SQL bytes escaping.
 func QuoteSQLBytes(bs []byte) string {
+	quote := suitableQuote(bs)
+
 	var buf bytes.Buffer
-	buf.WriteString("B\"")
+	buf.WriteString("b")
+	buf.WriteRune(quote)
 	for _, b := range bs {
-		q := quoteSingleEscape(rune(b))
+		q := quoteSingleEscape(quote, rune(b))
 		if q != "" {
 			buf.WriteString(q)
 			continue
 		}
-		if char.IsPrint(b) {
+
+		// Note: char.IsPrint(' ') is false
```

Comment:
I think it is better to fix `char.IsPrint`, isn't it?

Reply:
Yes. You could fix `char.IsPrint`.